src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
+
+  return sum/men.length;
```

Comment:
Assuming the above comments, we should refactor this line to something like this:


```js
    return totalAge/men.length;
```

We don't need to check if totalAge is a number, because the `reduce` method will return the initial value (in our case is `0`) if the reducer function returns `undefined`.


It's not considered a good practice, but i recommend storing this calculation in a variable, to achieve a more readable code.